### PR TITLE
Fix initialization of parameters in FSDP

### PR DESCRIPTION
**Pull request by awaelchli**
This patch ensures that the model gets initialized with random weights generated from the same seed across all processes. The seed is still randomized for data-loading to ensure we have different data shards per process.

### CODE REVIEW

**Comment by awaelchli** [comment]
> Shouldn't this be done to the finetuning scripts too?

Not necessary there, because they load the pretrained checkpoint file and so the random weights get overwritten.

**Review by carmocca — APPROVED** [review]

**Comment by carmocca** [comment]
But not the LoRA/Adapter weights. We'd want to do the same if we want them to be the same across processes

**Review by lantiga — APPROVED** [review]

**Comment by lantiga** [comment]
> But not the LoRA/Adapter weights. We'd want to do the same if we want them to be the same across processes

We can do it in a separate PR but I recommend doing it asap.
